Project.toml

```diff
@@ -17,6 +17,7 @@ Parameters = "d96e819e-fc66-5662-9728-84c9c7592b0a"
 Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
 Requires = "ae029012-a4dd-5104-9daa-d747884805df"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
+SplitApplyCombine = "03a91e81-4c3e-53e1-a0a4-9c0c8f19dd66"
```

Comment:
It is needed for this function:
```
# Creates a view of the pC vector at a given comaprtment.
function view_pC_vector(pC, comp, pC_location_types, pC_idxs)
    mapview(p_idx -> pC_location_types[p_idx] ? pC[p_idx][1] : pC[p_idx][comp], pC_idxs)
end
```
basically, we have parameters stored in a `Vector{Vector{Float64}}`. E.g. `p=[[1], [1,2,3],[1],[1,2,3]]` is  the value of the four parameters, across 3 compartments. The first and second parameter have the same value (`1`) across all compartments, and the other 2 have different values across the 3 compartments.
If I want to get the 2nd compartment value I want to create a view vector like `[1,2,1,2]`. I asked about the issue on the Slack and someone suggested this package.

Reply:
I think it would make more sense to cache a work vector in the function and just fill it in within this function via a loop. That might be more performant too.

Reply:
That also avoids a dependency on a package that doesn't seem super maintained from what I can see.

Reply:
It worked like this in a previous version of the PR, but I changed to this thinking it would be more performant. Could change back though.

Reply:
I have updated to using this function:
```julia
function view_pC_vector!(work_pC, pC, comp, enumerated_pC_idx_types)
    for (idx,loc_type) in enumerated_pC_idx_types
        work_pC[idx] = (loc_type ? pC[idx][1] : pC[idx][comp])
    end
    return work_pC
end
```
Performance is slightly reduced in some cases, and very slightly improved in others. I agree that reducing package dependency is good though. Here are benchmarks before the change:
```
Small grid, small, non-stiff, system. Runtime: 0.000272401, previous standard: 0.0006
Large grid, small, non-stiff, system. Runtime: 0.114815873, previous standard: 0.26
Small grid, small, stiff, system. Runtime: 0.026306312, previous standard: 0.17
Medium grid, small, stiff, system. Runtime: 0.532995638, previous standard: 2.3
Large grid, small, stiff, system. Runtime: 71.961035456, previous standard: 170.0
Small grid, mid-sized, non-stiff, system. Runtime: 0.001275261, previous standard: 0.0016
Large grid, mid-sized, non-stiff, system. Runtime: 0.570210118, previous standard: 0.67
Small grid, mid-sized, stiff, system. Runtime: 0.017001975, previous standard: 0.019
Large grid, mid-sized, stiff, system. Runtime: 91.262612505, previous standard: 35.0
```
and here after:
```
Small grid, small, non-stiff, system. Runtime: 0.000268705, previous standard: 0.0006
Large grid, small, non-stiff, system. Runtime: 0.112503055, previous standard: 0.26
Small grid, small, stiff, system. Runtime: 0.031937706, previous standard: 0.17
Medium grid, small, stiff, system. Runtime: 0.787503784, previous standard: 2.3
Large grid, small, stiff, system. Runtime: 84.327462341, previous standard: 170.0
Small grid, mid-sized, non-stiff, system. Runtime: 0.001234625, previous standard: 0.0016
Large grid, mid-sized, non-stiff, system. Runtime: 0.535502515, previous standard: 0.67
Small grid, mid-sized, stiff, system. Runtime: 0.014664551, previous standard: 0.019
Large grid, mid-sized, stiff, system. Runtime: 90.787294741, previous standard: 35.0
```
(please ignore the `previous standard: ...` part)

Reply:
Currently, SplitApplyCombine is not used, I will remove the dependency in the next commit.

Reply:
Now removed.